flamingo_tools/segmentation/unet_prediction.py

```diff
@@ -228,14 +243,45 @@ def run_unet_prediction(
     output_folder, model_path,
     min_size, scale=None,
     block_shape=None, halo=None,
+    prediction_instances=1,
+):
+    if prediction_instances > 1:
+        run_unet_prediction_slurm(
+            input_path, input_key, output_folder, model_path,
+            scale=scale, block_shape=block_shape, halo=halo,
+            prediction_instances=prediction_instances,
+        )
+    else:
+        os.makedirs(output_folder, exist_ok=True)
+
+        find_mask(input_path, input_key, output_folder)
+
+        original_shape = prediction_impl(
+            input_path, input_key, output_folder, model_path, scale, block_shape, halo
+        )
+
+        pmap_out = os.path.join(output_folder, "predictions.zarr")
+        segmentation_impl(pmap_out, output_folder, min_size=min_size, original_shape=original_shape)
+
+def run_unet_prediction_slurm(
+    input_path, input_key, output_folder, model_path,
+    scale=None,
+    block_shape=None, halo=None, prediction_instances=1,
 ):
     os.makedirs(output_folder, exist_ok=True)
+    prediction_instances = int(prediction_instances)
+    slurm_task_id = os.environ.get("SLURM_ARRAY_TASK_ID")
+    if slurm_task_id is not None:
```

Comment:
Is the case `slurm_task_id = None` well defined here? If not, we should raise some error if we reach this code block with this setting.